sky/data/skystorage.py

```diff
@@ -0,0 +1,250 @@
+"""Skystorage module"""
+import multiprocessing
+import os
+import pathlib
+import subprocess
+import time
+from typing import Any, List, Optional
+import urllib.parse
+
+import click
+import psutil
+
+from sky.utils import common_utils
+from sky.utils import db_utils
+
+_CSYNC_BASE_PATH = '~/.skystorage'
+
+_DB_PATH = os.path.expanduser('~/.skystorage/state.db')
+pathlib.Path(_DB_PATH).parents[0].mkdir(parents=True, exist_ok=True)
```

Comment:
Will let @Michaelvll comment on using a DB or not. However, if we do decide to use this should be under `~/.sky` to group all our metadata together.